packages/logs/src/domain/sender.ts

```diff
@@ -0,0 +1,64 @@
+import type { Context } from '@datadog/browser-core'
+import { combine, createContextManager, display, includes } from '@datadog/browser-core'
+import type { LogsMessage } from './logger'
+import { HandlerType, StatusType } from './logger'
+
+export type Sender = ReturnType<typeof buildSender>
+
+export const STATUS_PRIORITIES: { [key in StatusType]: number } = {
+  [StatusType.debug]: 0,
+  [StatusType.info]: 1,
+  [StatusType.warn]: 2,
+  [StatusType.error]: 3,
+}
+
+export function buildSender(
+  sendLog: (message: LogsMessage) => void,
+  handlerType: HandlerType | HandlerType[] = HandlerType.http,
+  level: StatusType = StatusType.debug,
+  loggerContext: Context = {}
+) {
+  const conf = {
+    level,
+    handlerType,
+  }
+  const contextManager = createContextManager()
+  contextManager.set(loggerContext)
+
+  function isAuthorized(status: StatusType, handlerType: HandlerType) {
+    const sanitizedHandlerType = Array.isArray(conf.handlerType) ? conf.handlerType : [conf.handlerType]
+    return STATUS_PRIORITIES[status] >= STATUS_PRIORITIES[conf.level] && includes(sanitizedHandlerType, handlerType)
+  }
+
+  function sendHttpRequest(message: string, messageContext: object | undefined, status: StatusType) {
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏Naming: maybe we could have `sendToHttp` / `sendToConsole`?

Reply:
Yes better